lib/internal/streams/legacy.js

```diff
@@ -111,4 +112,24 @@ function prependListener(emitter, event, fn) {
     emitter._events[event] = [fn, emitter._events[event]];
 }
 
+let defaultHighwaterMark = null;
+ObjectDefineProperty(Stream, 'defaultHighwaterMark', {
+  get () {
+    return defaultHighwaterMark || Buffer.poolSize || 16 * 1024;
+  },
+  set (value) {
+    defaultHighwaterMark = value;
```

Comment:
While we are here, we can check this to be a non-negative integer and throw otherwise.